### PR TITLE
Don't do comdb2_row* fields for synthetic tables

### PR DESCRIPTION
**Pull request by mponomar**
Also fixes a memory leak from dispatching to a prefault pool that doesn't exist.

### CODE REVIEW

**Comment by mponomar** [comment]
Akshat points out that I'm wrong about the memory leak.

File: sqlite/src/resolve.c [hunk 1]
```diff
@@ -415,11 +415,11 @@ static int lookupName(
     ** the vm sniffs this out when it runs OP_Rowid and executes the comdb2
     ** backend call to get the rrn+genid.
     */
-    else if( cnt==0 && cntTab==1 && sqlite3IsComdb2Rowid(zCol) ){
+    else if( cnt==0 && cntTab==1 && sqlite3IsComdb2Rowid(pMatch->pTab, zCol) ){
        cnt = 1;
```
Comment (mistachkin) [inline sqlite/src/resolve.c]: Upon closer examination, I think both changes here need "&& pMatch" just prior to the sqlite3IsComdb2*() calls...

Comment (mponomar) [reply sqlite/src/resolve.c]: Thanks!

**Comment by nirbhayc** [comment]
Could you also add a test please?

**Comment by mistachkin** [comment]
That's a good point, it would be nice to have test coverage for this.  Also, having a test where pMatch ends up being false (e.g. column not found?) would be nice as well.

**Comment by mponomar** [comment]
Ah, I had a test for this, that I somehow forgot to include (small comdb2sys change).

**Comment by adizaimi** [comment]
I think t7 and t8 subtests are not run anymore after this change.

**Pull request merged**